tests/v2/validation/workloads/reusable_workload_tests.go

```diff
@@ -0,0 +1,490 @@
+package workloads
+
+import (
+	"errors"
+	"fmt"
+
+	projectsapi "github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/workloads/cronjob"
+	"github.com/rancher/rancher/tests/v2/actions/workloads/daemonset"
+	"github.com/rancher/rancher/tests/v2/actions/workloads/deployment"
+	"github.com/rancher/rancher/tests/v2/actions/workloads/pods"
+	"github.com/rancher/rancher/tests/v2/actions/workloads/statefulset"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/workloads"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	kubectl_output_err = "No log output was recieved from kubectl"
+	nginxImage         = "nginx"
+	ubuntuImage        = "ubuntu"
+	redisImage         = "redis"
+	cronExpression     = "*/1 * * * *"
+)
+
+type WorkloadTestFunc func(client *rancher.Client, clusterID string) error
+
+type WorkloadTest struct {
+	name     string
+	testFunc WorkloadTestFunc
+}
+
+func GetAllWorkloadTests() []WorkloadTest {
+	tests := []WorkloadTest{
+		{"WorkloadDeploymentTest", WorkloadDeploymentTest},
+		{"WorkloadSideKickTest", WorkloadSideKickTest},
+		{"WorkloadDaemonSetTest", WorkloadDaemonSetTest},
+		{"WorkloadCronjobTest", WorkloadCronjobTest},
+		{"WorkloadStatefulsetTest", WorkloadStatefulsetTest},
+		{"WorkloadUpgradeTest", WorkloadUpgradeTest},
+		{"WorkloadPodScaleUpTest", WorkloadPodScaleUpTest},
+		{"WorkloadPodScaleDownTest", WorkloadPodScaleDownTest},
+		{"WorkloadPauseOrchestrationTest", WorkloadPauseOrchestrationTest},
+	}
+
+	return tests
+}
+
+// WorkloadDeploymentTest creates a deployment and validates that it is successfully deployed.
+func WorkloadDeploymentTest(client *rancher.Client, clusterID string) error {
+	log.Info("Creating a new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Creating a new deployment")
```

Comment:
Not a huge deal, but is there a specific reason why you have it as `log` and not just `logrus`? This question goes throughout this PR as well, just commenting it once here.